client/scripts/views/pages/validators/substrate/identity.ts

```diff
@@ -1,31 +1,104 @@
 import m from 'mithril';
+import app from 'state';
+import { Button, Classes, Icons, Icon } from 'construct-ui';
+import User from 'views/components/widgets/user';
+import { ChainBase } from 'models';
+import { truncate } from 'lodash';
+import { externalLink } from 'helpers';
+import Substrate from 'controllers/chain/substrate/main';
+import { makeDynamicComponent } from 'models/mithril';
 import { IAccountInfo } from 'controllers/chain/substrate/staking';
 
-const Identity: m.Component<IAccountInfo, {}> = {
+const truncLength = {
+  length: 20
+};
+
+export interface IValidatorState {
+  dynamic: {
+    info: IAccountInfo;
+  },
+  isNominating: boolean;
+}
+
+export interface IdentityAttrs {
+  stash: string;
+}
+
+const Identity = makeDynamicComponent<IdentityAttrs, IValidatorState>({
+  getObservables: (attrs) => ({
+    // we need a group key to satisfy the dynamic object constraints, so here we use the chain class
+    groupKey: app.chain.class.toString(),
```

Comment:
`groupKey` should be set to `attrs.stash` or else the observable wont change/update if the stash value is updated.